src/screens/SSIContactAddScreen/index.tsx

```diff
@@ -54,26 +54,37 @@ class SSIContactAddScreen extends PureComponent<IProps, IState> {
   };
 
   onCreate = async (): Promise<void> => {
-    const {identities, name, uri, onCreate} = this.props.route.params;
-    const {contactAlias} = this.state;
-
     Keyboard.dismiss();
 
-    this.onValidate(contactAlias)
-      .then(() => {
-        this.props.createContact({
-          name,
-          alias: contactAlias.trim(),
-          uri,
-          identities,
-        });
-        onCreate();
+    this.onValidate(this.state.contactAlias)
+      .then(async () => {
+        await this.upsert();
+        await this.props.route.params.onCreate();
```

Comment:
i rather deconstruct these things. so

const { onCreate } = this.props.route.params

and then call await onCreate()

keeps the code a lot cleaner and more readable if we get more

Reply:
changed.